File: src/test/java/etlmail/front/gui/sendmail/NewsletterNotificationBuilderTest.java
package etlmail.front.gui.sendmail;

import org.fest.assertions.Assertions;
import org.junit.Test;

public class NewsletterNotificationBuilderTest {
    @Test
    public void lesChampsDuFormulaireNeSontJamaisNuls() {
	NewsletterNotificationBuilder builder = new NewsletterNotificationBuilder();

	Assertions.assertThat(builder.to()).as("to").isNotNull();
	Assertions.assertThat(builder.cc()).as("cc").isNotNull();
	Assertions.assertThat(builder.from()).as("from").isNotNull();
	Assertions.assertThat(builder.subject()).as("subject").isNotNull();
	Assertions.assertThat(builder.template()).as("template").isNotNull();
    }
}

File: src/test/scala/etlmail/front/gui/sendmail/NewsletterNotificationBuilderTest.scala
package etlmail.front.gui.sendmail

import org.scalatest.matchers.ShouldMatchers
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.junit.runner.RunWith

@RunWith(classOf[JUnitRunner])
class NewsletterNotificationBuilderTest extends FlatSpec with ShouldMatchers {
  "the form's fields" should "never be null" in {
    val builder = new NewsletterNotificationBuilder

    builder.to should not be null
    builder.cc should not be null
    builder.from should not be null
    builder.subject should not be null
    builder.template should not be null
  }
}
